File: #100.Same Tree/Same Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {

    if(p==null || q==null){
        return p==q;
    }

    return p.val == q.val && arguments.callee(p.left,q.left) && arguments.callee(p.right,q.right);
};

File: #100.Same Tree/Same Tree.cpp
/*
判断两个二叉树是否相等的递归算法
*/
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
        if(p==NULL || q==NULL)
            return p==q;
        return isSameTree(p->left,q->left) && isSameTree(p->right,q->right) && p->val==q->val;
    }
};